account_invoice_validation_purchase_type/models/account_move.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.move"
+
+    product_type_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Purchase type",
+        help="Technical field used for the Invoice validation widget",
+        domain=[("purchase_type", "=", True), ("active", "=", True)],
+    )
+
+    def update_move_line_with_product_type(self):
+        for move in self:
+            product = move.product_type_id
+            taxes = product.taxes_id
+            expense_account = product.property_account_expense_id
+            lines = move.invoice_line_ids.filtered(
+                lambda l: l.display_type == "product"
+            )
+            for line in lines:
+                if line.move_type in ("in_invoice", "in_refund"):
+                    name = line.name
+                    line.write(
+                        {
+                            "product_id": product.id,
+                            "name": name,
+                            "tax_ids": taxes.ids,
+                            "account_id": expense_account.id,
+                        }
+                    )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "product_type_id" in vals:
```

Comment:
@acsonefho First somebody create the account.move, then another employee must validate the account.move and attach a purchase type to the account.move, so that can't be done in a create function as the account.move is already created when we attach a purchase type